### DISCUSSION TITLE
Fetch latest record / entry in Deta base?

### DISCUSSION THREAD

**Discussion by TychoLex42** [deta/cloud-docs#344] [Help] [answered]
Quick question: How would I retrieve the latest entry/record in the Deta Base? Based on a date is not possible, right? Or based on the key? Sorry for being fairly new to this! Thanks so much for helping me and building this awesome tool!

**Comment by xeust** [comment, marked as answer]
Hi @TychoLex42 , Base will return items using `fetch` in ascending order, so you can indeed do this, using a variation of `unix timestamps` as keys.

If you simply use `unix timestamps` as keys, this will do the opposite of what you want -- as the first item you fetch will be the oldest item (smallest time stamp).

So what you can do, is use keys that that get smaller as unix time gets larger. So if you store items with the key scheme `some really big number` - `unix time stamp`, the first item you get from a fetch will be the most recent.

Hope that helps!

**Reply by TychoLex42** [reply]
Hi @xeust,
thanks for your quick answer. I have tried to implement your approach. If there are multiple entries in the db, the value for `some really big number` would have to change as well, e.g. ascending for each new entry, am I correct? Otherwise, the entry with the key `some really big number - unix time stamp` with the oldest time stamp will again be the first that is fetched.
The changing of the big number will depend however on the previous entry, so that I would have to do a fetch of the db with the current first, right?
I guess there is no parameter to change the fetching in a descending order, right?
Thanks so much for your help!

**Reply by xeust** [reply]
Hi @TychoLex42 ,

You'll actually want to keep the `bigNumer` constant -- this guarantees that your keys will be getting **_smaller_** the later you store them (since `fetch` returns in ascending order). If you increase `bigNumber` with each new item, you'd break the guarantee that keys get smaller with time.

Here's a simple express server that should do what you want:

```javascript
const express = require('express');
const { Deta } = require('deta');

const app = express(); 

const deta = Deta(); 

const db = deta.Base('simple_db'); 

const bigNumber = 8.64e15;

const generateKey = (timestamp) => {
	return `${ bigNumber -  timestamp}`
}

app.get('/', async (req, res) => {
    const timestamp = Date.now();
    const data = {'key': generateKey(timestamp), 'timestamp': Date(timestamp).toString()}
    await db.put(data);
    res.send('Hello World!');
})

module.exports = app;
```

Each time you hit the endpoint, a new item is stored in your Base under the key scheme of `bigNumer - Date.now()`. I've also stored `Date.now()` in human readable format next to the key. If you load up your Base's UI, you'll see that the most recent item stored (expand the `timestamp` field) is always the first item returned via `fetch`:

<img width="578" alt="Screen Shot 2022-07-28 at 12 14 43" src="https://user-images.githubusercontent.com/7769878/181481696-932e3e9f-9cae-496c-bb46-c11b2868cf10.png">

(One small note -- if you're expecting to store multiple items in the same millisecond, you'll want to append the key scheme with a bit more information to prevent a collision of keys).

Please let me know if you can't get what you want working.

**Reply by TychoLex42** [reply]
Hey @xeust,
thanks so much again for your long and detailed reply.
I now know what I understood wrong in your previous answer. I thought that your suggestion of `some really big number - unix time stamp` was meant to be a string, e.g. "9999 - 2022-07-28 15:13". But I know get that you suggested to subtract the timestamp from the large number! My logic was flawed and I should have read your answer more carefully!
I know get it that the large number has to stay the same, as the time stamp gets "higher".
Thanks again for your help! :-)

**Reply by xeust** [reply]
Hi @TychoLex42 -- no problem :)

**Reply by chandrashekarcn** [reply]
make one entry with key new_rec and every time you put data in base updte new_rec with that key. 

ex: 
create one record as below
{key:new_rec,last:timestamp}

every time when you put deta in detabase update{key:new_rec, last:timestamp(new entry) }

**Reply by Sokhavuth** [reply]
I would pull all documents from Deta Base first. The result is a JS array of objects. So we can use vanilla JavaScript sort() method to sort this array of objects by date. The result we can get the newest or the oldest record by date.
```js
let result = await mydb.posts.fetch(query)
let allItems = result.items

while(result.last){
       result = await mydb.posts.fetch(query, {last: result.last})
       allItems = allItems.concat(result.items)
}

allItems.sort((a, b) => {     
    let da = new Date(a.date)
    let db = new Date(b.date)
    return db - da
})
        
const lastItem = allItems[allItems.length-1]
```
